OpenTabletDriver.Web/Views/Wiki/FAQ/Linux.md

```diff
@@ -0,0 +1,241 @@
+
+This issue can be caused by either your initramfs not containing the instructions to not load the modules
+or another crucial service loading the module despite it being blacklisted. You will have to find that service
+yourself if the solution below does not solve your issue.
+
+Rebuilding the initramfs is an easy process, but the commands to do this vary between the distro.
+
+#### Arch
+
+```bash
+sudo mkinitcpio -P
+```
+
+#### Debian / Ubuntu
+```bash
+sudo update-initramfs -u
+```
+
+### My tablet detects, but it isn't working? {#fail-virtual-device}
+
+If this is on a fresh install of OpenTabletDriver, usually problems like this can be solved
+simply by restarting your computer. If it isn't resolved you may try resetting your OpenTabletDriver
+settings to the defaults with `File > Reset to defaults`, then pressing save.
+
+If your issue is still not resolved, and you are using OpenTabletDriver.UX, navigate to the "Console" tab
+at the top of the main interface, switch from `Info` to `Debug` in the bottom left of that window and you will be see every log message.
+If you are using the daemon directly you may simply just look for a log
+that resembles the following:
+
+#### EACCESS
+
+> Failed to initialize virtual tablet. (error code EACCES)
+
+This usually occurs when udev is improperly set up or hasn't explictly been reloaded.
+
+Make sure your rules are correct by heading [here](#udev).
+
+#### ENODEV
+
+> Failed to initialize virtual tablet. (error code ENODEV)
+
+This error occurs after updating the kernel but without rebooting, you should reboot your computer
+and check if the problem is resolved.
+
+#### ENOENT
+This error occurs when the `CONFIG_INPUT_UINPUT` kernel option is disabled. Make sure it is enabled in
+your kernel config.
+
+##### Gentoo
+Check the option `CONFIG_INPUT_UINPUT` in /usr/src/linux.config. If it is not enabled, use the following steps
+
+```bash
+cd /usr/src/linux
+sudo make menuconfig
+```
+
+Having done that, select y (for always active) or m (for building as a module) for the option in
+*Device Drivers ---> Input Device Support ---> Miscellaneous devices ---> User level driver support* and rebuild the kernel.
+
+##### Other Distros
+
+Refer to your distribution's documentation regarding kernel configuration to enable this option.
+
+This is probably due to your kernel not having uinput built in, if `stat /dev/uinput` doesn't return a file and you are using a custom kernel,
+compile your kernel with support for uinput or use a different kernel.
+
+### Poor performance with NVIDIA {#performance-nvidia}
+
+Disable "Force full composition pipeline" in the NVIDIA settings panel.
+
+### Stuck cursor in osu!lazer (Wayland) {#osu-lazer-broken-input-wayland}
+
+Make sure you set the `SDL_VIDEODRIVER` to `wayland`:
+
+```bash
+env SDL_VIDEODRIVER=wayland ./osu.AppImage
+```
+or
+```bash
+env SDL_VIDEODRIVER=wayland osu-lazer
+```
+
```

Comment:
```suggestion
You can use either of these commands when creating a shortcut, script, or `.desktop` file to run osu! with, or add it to the existing runner you're already using to launch osu!.
```

Reply:
Suggesting more than provided seems out of scope, since there is a chance users might misunderstand that they can freely edit system files (which the osu-lazer desktop file can be), undoing their changes eventually on a system update.

Reply:
I agree. We shouldn't suggest modifying system files.